### PR TITLE
[NFC] Group all `-preview=/-revert=` options in `global.params`

### PR DESCRIPTION
**Pull request by thewilsonator**
If we want to have preview switches be per module/package/whatever we will need to do something with them, so let's start by grouping them all together.

### CODE REVIEW

File: compiler/src/dmd/globals.d [hunk 1]
```diff
@@ -131,31 +127,10 @@ extern (C++) struct Param
...
     bool allInst;           // generate code for all template instantiations
-    bool fix16997 = true;   // fix integral promotions for unary + - ~ operators
-                            // https://issues.dlang.org/show_bug.cgi?id=16997
-    bool fixAliasThis;      // if the current scope has an alias this, check it before searching upper scopes
-    bool inclusiveInContracts;   // 'in' contracts of overridden methods must be a superset of parent contract
-    /** The --transition=safe switch should only be used to show code with
-     * silent semantics changes related to @safe improvements.  It should not be
-     * used to hide a feature that will have to go through deprecate-then-error
-     * before becoming default.
-     */
```
Comment (thewilsonator) [inline compiler/src/dmd/globals.d:148]: `vsafe` was removed a long time ago.

File: compiler/src/dmd/globals.d [hunk 2]
```diff
@@ -173,6 +148,28 @@ extern (C++) struct Param
     bool hcUsage;           // print help on -HC switch
     bool logo;              // print compiler logo
 
+    // Options for `-preview=/-revert=`
+    FeatureState useDIP25;       // implement https://wiki.dlang.org/DIP25
+    FeatureState useDIP1000;     // implement https://dlang.org/spec/memory-safe-d.html#scope-return-params
+    bool ehnogc;                 // use @nogc exception handling
+    bool useDIP1021;             // implement https://github.com/dlang/DIPs/blob/master/DIPs/accepted/DIP1021.md
```
Comment (thewilsonator) [inline compiler/src/dmd/globals.d:155]: I feel these `bool`s should all be `FeatureState`s, but I'll leave this to a separate PR.

**Comment by ljmf00** [comment]
We have a problem porting all the preview options into globals. We might want to think of a different approach, if we want to support per module preview switches. I talked with @atilaneves at DConf about this.

**Comment by thewilsonator** [comment]
Not really, we should reuse the machinery for `-mv=`, i.e. have `-preview=option[={module spec}]_opt`. 
The only thing that gets weird is what happens with templates and what modules things actually belong to. 

However all of that has nothing to do with grouping these together for easier code navigation.

**Comment by ljmf00** [comment]
> However all of that has nothing to do with grouping these together for easier code navigation.

Yeah, I just wanted to note this, because we might want to restructure it to support that feature.

**Review by ibuclaw — APPROVED** [review]

**Pull request merged**